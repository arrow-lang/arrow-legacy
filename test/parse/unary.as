# Promote
+32;
+++643;

# Negate (Numeric)
-32;
--0xafe;

# Negate (Bitwise)
!6;

# Negate (Logical)
not false;
