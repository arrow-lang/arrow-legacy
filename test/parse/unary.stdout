<?xml version="1.0" encoding="utf-8"?>
<Module>
 <Promote>
  <Integer base="10">32</Integer>
 </Promote>
 <Promote>
  <Integer base="10">643</Integer>
 </Promote>
 <NegateNumeric>
  <Integer base="10">32</Integer>
 </NegateNumeric>
 <NegateNumeric>
  <Integer base="16">afe</Integer>
 </NegateNumeric>
 <NegateBitwise>
  <Integer base="10">6</Integer>
 </NegateBitwise>
 <NegateLogical>
  <Boolean>false</Boolean>
 </NegateLogical>
</Module>
